### PR TITLE
RFR: email to username

### PR DESCRIPTION
**Pull request by melodyfelix**
## Status: 
🚀 rfr

## Description
- Changed user model to take in username instead of email
- Changed all endpoints to account for email/username change
- Changed login/register page to take in username instead of email
- Fixed update profile to work properly with revised endpoint
- Fixed successfully spelling error

Related issues: #84 #176

### CODE REVIEW

File: backend/api/views/user.py [hunk 1]
```diff
@@ -25,20 +25,20 @@ def edit_user():
             message="User not found", status=400, data={"status": "fail"}
         )
 
-    # name or email not included in json
-    if ("name" not in user_data) or ("email" not in user_data):
+    # name or username not included in json
```
Comment (micleo2) [inline backend/api/views/user.py]: :lock: good work

File: backend/tests/test_authenticate.py [hunk 2]
```diff
@@ -35,27 +35,27 @@ def test_register(client):
...
+    user = User(name="bob", password="password123", username="test@gmail.com")
 
     db.session.add(user)
     db.session.commit()
 
     login_res = client.post(
         "/login",
         data=json.dumps(
-            dict(name="bob", password="password123", email="test@gmail.com")
+            dict(name="bob", password="password123", username="test@gmail.com")
```
Comment (micleo2) [inline backend/tests/test_authenticate.py]: interesting how u used username instead of email here

Comment (micleo2) [reply backend/tests/test_authenticate.py]: :lock:

File: frontend/src/components/Register.js [hunk 3]
```diff
@@ -22,7 +21,7 @@ class Register extends Component {
       name: '',
       password: '',
       password_confirm: '',
-      email: '',
+      username: '',
```
Comment (micleo2) [inline frontend/src/components/Register.js]: The first PR in history to include FE and BE changes :heavy_check_mark: :lock:

Comment (XuanLi-Leong) [reply frontend/src/components/Register.js]: L E G E N D @melodyfelix

File: frontend/src/pages/AccountManagePage.js [hunk 4]
```diff
@@ -63,16 +62,33 @@ class AccountManagePage extends Component {
...
+      username: this.state.username
     };
-    console.log(postUserData(userData));
+
+    if (
+      this.state.changedUsername.length > 0 &&
+      this.state.changedName.length > 0
+    ) {
+      userData = {
+        name: this.state.changedName,
```
Comment (micleo2) [inline frontend/src/pages/AccountManagePage.js]: Squash :clap: and merge :twisted_rightwards_arrows:

**Pull request merged**